apt-hook/hook.cc

```diff
@@ -201,69 +282,230 @@ int main(int argc, char *argv[])
-                           "%d of the updates are from UA Infra: ESM.",
-                           res.enabled_esms_i),
-                  res.enabled_esms_i);
-         ioprintf(std::cout, "\n");
+   // Compute all strings necessary to fill in templates
+   std::string space_separated_esm_i_packages = "";
+   if (res.esm_i_packages.size() > 0) {
+      for (uint i = 0; i < res.esm_i_packages.size() - 1; i++) {
+         space_separated_esm_i_packages.append(res.esm_i_packages[i]);
+         space_separated_esm_i_packages.append(" ");
       }
-      if (res.enabled_esms_a > 0)
-      {
-         ioprintf(std::cout,
-                  ngettext("%d of the updates is from UA Apps: ESM.",
-                           "%d of the updates are from UA Apps: ESM.",
-                           res.enabled_esms_a),
-                  res.enabled_esms_a);
-         ioprintf(std::cout, "\n");
+      space_separated_esm_i_packages.append(res.esm_i_packages[res.esm_i_packages.size() - 1]);
+   }
+   std::string space_separated_esm_a_packages = "";
+   if (res.esm_a_packages.size() > 0) {
+      for (uint i = 0; i < res.esm_a_packages.size() - 1; i++) {
+         space_separated_esm_a_packages.append(res.esm_a_packages[i]);
+         space_separated_esm_a_packages.append(" ");
       }
+      space_separated_esm_a_packages.append(res.esm_a_packages[res.esm_a_packages.size() - 1]);
    }
-
-   if (res.disabled_esms_i > 0 || res.disabled_esms_a > 0)
-   {
-      if (command_used != "update")
-         std::cout << std::endl;
-      if (res.disabled_esms_i > 0)
+   std::string esm_a_packages_count = std::to_string(res.esm_a_packages.size());
+   std::string esm_i_packages_count = std::to_string(res.esm_i_packages.size());
+
+   // Execute specified subcommand
+   if (subcommand == ProcessTemplates) {
+      std::array<std::string, 4> template_file_names = {
+         CONTRACT_EXPIRED_APT_UPGRADE_MESSAGE_TEMPLATE_PATH,
+         CONTRACT_EXPIRED_APT_DIST_UPGRADE_MESSAGE_TEMPLATE_PATH,
+         ESM_APPS_NOT_ENABLED_MESSAGE_TEMPLATE_PATH,
+         ESM_INFRA_NOT_ENABLED_MESSAGE_TEMPLATE_PATH
+      };
+      std::array<std::string, 4> static_file_names = {
+         CONTRACT_EXPIRED_APT_UPGRADE_MESSAGE_STATIC_PATH,
+         CONTRACT_EXPIRED_APT_DIST_UPGRADE_MESSAGE_STATIC_PATH,
+         ESM_APPS_NOT_ENABLED_MESSAGE_STATIC_PATH,
+         ESM_INFRA_NOT_ENABLED_MESSAGE_STATIC_PATH
+      };
+      for (uint i = 0; i < template_file_names.size(); i++) {
+         process_template_file(
+            template_file_names[i], 
+            static_file_names[i], 
+            esm_a_packages_count,
+            space_separated_esm_a_packages,
+            esm_i_packages_count,
+            space_separated_esm_i_packages
+         );
+      }
+   } else if (subcommand == PostInvokeStats || subcommand == PostInvokeSuccess || test_run) {
+      // TODO this was the existing functionality - how much of this are we keeping?
```

Comment:
Just calling out this comment. This should work the same as before currently for apt update